client-src/elements/chromedash-feature-row.ts

```diff
@@ -1,34 +1,35 @@
 import {LitElement, css, html, nothing} from 'lit';
+import {customElement, property} from 'lit/decorators.js';
 import {SHARED_STYLES} from '../css/shared-css.js';
+import {Feature, StageDict} from '../js-src/cs-client.js';
 import {
+  GATE_ACTIVE_REVIEW_STATES,
   GATE_TEAM_ORDER,
   STAGE_SHORT_NAMES,
-  GATE_ACTIVE_REVIEW_STATES,
 } from './form-field-enums';
+import {GateDict} from './chromedash-gate-chip.js';
 
-class ChromedashFeatureRow extends LitElement {
-  static get properties() {
-    return {
-      feature: {type: Object},
-      columns: {type: String},
-      signedIn: {type: Boolean},
-      canEdit: {type: Boolean},
-      starredFeatures: {type: Object},
-      approvals: {type: Object},
-      gates: {type: Object},
-      selectedGateId: {type: Number},
-    };
-  }
+interface ActiveStagesAndGates {
+  stage: StageDict;
+  gates: any[]; // TODO(markxiong0122): Add Gate type when PR#4060 is merged.
+}
 
-  constructor() {
-    super();
-    this.starredFeatures = new Set();
-    this.feature = null;
-    this.canEdit = false;
-    this.approvals = {};
-    this.gates = {};
-    this.selectedGateId = 0;
-  }
+@customElement('chromedash-feature-row')
+class ChromedashFeatureRow extends LitElement {
+  @property({type: Object, attribute: false})
+  feature!: Feature;
+  @property({type: String})
+  columns;
+  @property({type: Boolean})
+  canEdit = false;
+  @property({type: Boolean})
+  signedIn;
```

Comment:
Remember to add types and initializers to fields. I think these are:
```suggestion
  @property({type: String})
  columns: undefined | 'approvals';
  @property({type: Boolean})
  canEdit = false;
  @property({type: Boolean})
  signedIn = false;
```